File: working/plugins/hasher.lua
local input = Aurora.box("")
local output = 0
local cacheHit = false
local rhashHit = nil

return {
	id = "aurora.hasher",
	gui = {
		visible = true,
		title = "Hasher",
		OnGui = function()
			if ImGui.InputText("Input", input) then
				local unescaped = Aurora.unescape(Aurora.unbox(input))
				output = Aurora.hash(unescaped)
				cacheHit = Aurora.cache_hit(string.format("%x.pc", output))
				rhashHit = Aurora.rhash(output)
			end
			ImGui.Text("0x%x", output)

			if cacheHit then
				ImGui.Text("File found: %x.pc", output)
			end

			if rhashHit then
				ImGui.Text("Result found in hashtable: %s", rhashHit)
			end
		end
	},
}

File: java/src/xyz/anthofoxo/aurora/gui/Hasher.java
package xyz.anthofoxo.aurora.gui;

import imgui.ImGui;
import imgui.flag.ImGuiCond;
import imgui.type.ImBoolean;
import imgui.type.ImString;
import xyz.anthofoxo.aurora.Hash;

public class Hasher {
	public static String commExport;

	public ImBoolean visible = new ImBoolean();
	public ImString input = new ImString(512);
	public Integer hash;
	public String output;
	public boolean collision;

	public Hasher() {
		compute();
	}

	public void compute() {
		hash = Hash.fnv1a(Hash.escapedToByteArray(input.get()));
		output = Integer.toHexString(hash);
		collision = Hash.hashes.containsKey(hash);
	}

	public void draw() {
		if (!visible.get()) return;

		ImGui.setNextWindowSize(640, 480, ImGuiCond.FirstUseEver);
		if (ImGui.begin("Hasher", visible)) {
			if (ImGui.button("Reload Hashes")) {
				Hash.reloadHashes();
				compute();
			}

			if (ImGui.inputText("Input", input)) compute();
			ImGui.labelText("Output", output);

			if (collision) {
				ImGui.text("Hash found");

				if (ImGui.button("Open in ObjLib parse")) {
					commExport = output;
				}
			}
		}
		ImGui.end();
	}
}
